vm/src/obj/objstr.rs

```diff
@@ -149,13 +148,12 @@ impl PyValue for PyStringReverseIterator {
 impl PyStringReverseIterator {
     #[pymethod(name = "__next__")]
     fn next(&self, vm: &VirtualMachine) -> PyResult {
-        if self.position.get() > 0 {
-            let position: usize = self.position.get() - 1;
+        let pos = self.position.load();
```

Comment:
Again. Use `fetch_add`

Reply:
This is a reverse iterator, though, and fetch_sub (and fetch_add) wrap on over/underflow, and doing a fetch_op before the if means it's unconditional, so if you call `next(it)`  after it does a StopIteration than it would go to usize::max and be out of bounds.